### DISCUSSION TITLE
Filtering of response messages or codes?

### DISCUSSION THREAD

**Discussion by SRZEUS1990** [projectdiscovery/nuclei#5088] [Ideas]
### Please describe your feature request:
Filter the response if it contains the same message or response code.  This could be useful for filtering out responses. 

### Describe the use case of this feature:
Adding a functionality that examines incoming responses for similarities to previously received responses. If an answer is deemed too similar to an existing one, it is filtered or tagged for review, hence improving the system's response quality and diversity.

**Comment by GeorginaReeder** [comment]
Thanks so much for this suggestion @SRZEUS1990 , we appreciate it!

We also have a [Discord server](https://discord.com/invite/projectdiscovery) that you're welcome to join. It's a great place to connect with fellow contributors and stay updated with the latest developments!

**Comment by ehsandeep** [comment]
@SRZEUS1990 feature request / use case is unclear, could you detail this out further what you are looking to achieve here to better understand the feature request?

**Reply by SRZEUS1990** [reply]
Drawback is that the tool is have no flags to identify whether the webpage response message changes or stays the same while it is being scanned using these templates.

You can only view the web page real message by manually clicking on the link.


Imagine you're using Neclei to scan a website for different URL endpoints. The tool detects multiple endpoints, let's say:

    /homepage
    /products
    /about

Neclei identifies these endpoints, but it doesn't provide information on whether the responses from each of these endpoints are the same or different. 

Neclei helps identify endpoints and other issues, but it doesn't automatically reveal if the responses from those endpoints are same or different. This evaluation requires manual inspection.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
